processing/src/main/java/io/druid/segment/data/GenericIndexedWriter.java

```diff
@@ -82,6 +132,11 @@ public void write(T objectToWrite) throws IOException
     valuesOut.write(bytesToWrite);
 
     headerOut.write(Ints.toByteArray((int) valuesOut.getCount()));
+    headerOutLong.write(Longs.toByteArray(valuesOut.getCount()));
```

Comment:
can we avoid writing headerOutLong file in cases where its not needed? may be, we can write headerOutLong on demand at the point when fileSizeLimit is crossed (and stop writing headerOut afterwards)? in most use cases, headerOutLong would not need to be written at all.